docs/02_cleos/03_command-reference/validate/signatures.md

```diff
@@ -0,0 +1,46 @@
+## Description
+Validate signatures and recover public keys
+
+## Positional Parameters
+- `transaction` _TEXT_ - The JSON string or filename defining the signed transaction to validate
+
+## Options
+- `-c,--chain-id` - The chain id that will be used in signature verification
+
+## Example
+<!--
+{
+  "expiration": "2020-04-23T04:47:23",
+  "ref_block_num": 20,
+  "ref_block_prefix": 3872940040,
+  "max_net_usage_words": 0,
+  "max_cpu_usage_ms": 0,
+  "delay_sec": 0,
+  "context_free_actions": [],
+  "actions": [
+    {
+      "account": "eosio",
+      "name": "voteproducer",
+      "authorization": [
+        {
+          "actor": "initb",
+          "permission": "active"
+        }
+      ],
+      "data": "000000008093dd74000000000000000001000000008093dd74"
+    }
+  ],
+  "transaction_extensions": [],
+  "signatures": [
+    "SIG_K1_Jy81u5yWSE4vGET1cm9TChKrzhAz4QE2hB2pWnUsHQExGafqhVwXtg7a7mbLZwXcon8bVQJ3J5jtZuecJQADTiz2kwcm7c"
+  ],
+  "context_free_data": []
+}
+-->
+```sh
+cleos -u https://api.testnet.eos.io validate signatures '{ "expiration": "2020-04-23T04:47:23", "ref_block_num": 20, "ref_block_prefix": 3872940040, 
+"max_net_usage_words": 0, "max_cpu_usage_ms": 0, "delay_sec": 0, "context_free_actions": [], "actions": [ { "account": "eosio", "name": "voteproducer", "authorization": [ { "actor": "initb", "permission": "active" } ], "data": "000000008093dd74000000000000000001000000008093dd74" } ], "transaction_extensions": [], "signatures": [ "SIG_K1_Jy81u5yWSE4vGET1cm9TChKrzhAz4QE2hB2pWnUsHQExGafqhVwXtg7a7mbLZwXcon8bVQJ3J5jtZuecJQADTiz2kwcm7c" ], "context_free_data": [] }'
+```
+
```

Comment:
Do you not want to put the JSON output of this example here?
```json
[
  "EOS7pCywBCz5zw2bc7teCVcT7MEWUr9s749qnYDNPEsBoH32vGqqN"
]
```

Reply:
Yes indeed, I just added the output. I also missed the transaction keyword and removed the url, which was unnecessary.

Reply:
It isn't a `transaction` keyword. That is just a placeholder for the actual transaction. Now that you added the word `transaction` the command no longer works.

Also, the a URL to a valid nodeos API node is necessary unless you specify the optional `--chain-id` parameter.

So I would recommend either keeping that URL or alternatively adding `--chain-id cf057bbfb72640471fd910bcb67639c22df9f92470936cddc1ade0e2f2e7dc4f` to the example command.

Reply:
Ah sorry I will fix it back - just realized `transaction` is used as an option not a subcommand as in `cleos sign transaction` or `cleos push transaction`. I will add both examples, w/ and w/o the chain id.

Reply:
Done